praw/reddit.py

```diff
@@ -741,6 +741,8 @@ def request(
                     "Unexpected BadRequest without json body. Please file a "
                     "bug at https://github.com/praw-dev/praw/issues"
                 ) from exception
+            if list(data) == ["message", "error"]:
```

Comment:
```suggestion
            if set(data) == {"error", "message"}:
```

This approach is a bit more robust as it doesn't depend on the iteration order.